src/Aspire.Hosting/Dashboard/ResourceSnapshot.cs

```diff
@@ -26,8 +25,9 @@ internal abstract class ResourceSnapshot
     public required ImmutableArray<EnvironmentVariableSnapshot> Environment { get; init; }
     public required ImmutableArray<VolumeSnapshot> Volumes { get; init; }
     public required ImmutableArray<UrlSnapshot> Urls { get; init; }
-    public required HealthStateKind? HealthState { get; set; }
+    public required ImmutableArray<HealthReportSnapshot> HealthReports { get; init; }
```

Comment:
We want the aggregate state top level as well as exposing individual heath states.